apps/admin/frontend/src/components/save_backend_file_modal.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { useContext, useState } from 'react';
+import styled from 'styled-components';
+import { basename, join } from 'path';
+import {
+  assert,
+  throwIllegalValue,
+  assertDefined,
+  Result,
+  sleep,
+} from '@votingworks/basics';
+import {
+  isElectionManagerAuth,
+  isSystemAdministratorAuth,
+  usbstick,
+} from '@votingworks/utils';
+import type { ExportDataError } from '@votingworks/admin-backend';
+
+import {
+  Button,
+  Modal,
+  H1,
+  P,
+  useExternalStateChangeListener,
+  Font,
+} from '@votingworks/ui';
+
+import { AppContext } from '../contexts/app_context';
+import { Loading } from './loading';
+
+export const UsbImage = styled.img`
+  margin-right: auto;
+  margin-left: auto;
+  height: 200px;
+`;
+
+interface SaveAsButtonProps {
+  onSave: (location: string) => void;
+  options?: KioskBrowser.SaveDialogOptions;
+}
+
+export function SaveAsButton({
+  onSave,
+  options,
+}: SaveAsButtonProps): JSX.Element {
+  async function useSaveDialog() {
+    assert(window.kiosk);
+    const { filePath } = await window.kiosk.showSaveDialog(options);
+    if (filePath) {
+      onSave(filePath);
+    }
+  }
+  return <Button onPress={useSaveDialog}>Save As…</Button>;
+}
+
+export type SaveBackendFileError =
+  | ExportDataError
+  | {
+      type: 'api-error';
+      message: string;
+    };
+
+export type SaveBackendFileResult = Result<string[], SaveBackendFileError>;
+
+export interface SaveBackendFileModalProps {
+  onSave: (location: string) => Promise<SaveBackendFileResult>;
+  onClose: () => void;
+  /**
+   * Title case name for the type of file being saved, e.g. "Batch Results".
+   */
+  fileTypeTitle: string;
+  /**
+   * Lowercase name for the type of file being saved, e.g. "batch results".
+   */
+  fileType: string;
+  /**
+   * Relative to the root of the USB drive.
+   */
+  defaultRelativePath: string;
+}
+
+type SaveFileModalState = 'error' | 'saving' | 'done' | 'init';
+
+export function SaveBackendFileModal({
+  onSave,
+  onClose,
+  fileTypeTitle,
+  fileType,
+  defaultRelativePath,
+}: SaveBackendFileModalProps): JSX.Element {
+  const { usbDrive, auth } = useContext(AppContext);
+  assert(isElectionManagerAuth(auth) || isSystemAdministratorAuth(auth));
+
+  const [currentState, setCurrentState] = useState<SaveFileModalState>('init');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [usbDrivePath, setUsbDrivePath] = useState<string>();
+
+  useExternalStateChangeListener(usbDrive.status, async (newStatus) => {
+    if (newStatus === 'mounted') {
+      const usbDriveInfo = await usbstick.getInfo();
+      assert(usbDriveInfo);
+      setUsbDrivePath(usbDriveInfo.mountPoint);
+    }
+  });
+
+  async function saveFile(location: string) {
+    setCurrentState('saving');
+
+    // Wait at least 1 second before changing the state to "done" or "error".
```

Comment:
I originally stripped it because I thought it unnecessary but prefer the UX with a bit of a pause. Added a comment in adb1b9434:
```
    // Wait at least 1 second before changing the state to "done" or "error". Otherwise
    // the "Saving…" message might flash very quickly on the screen, giving the user the
    // impression they missed something.
```

Reply:
Hm, not a big fan of that as a pattern - I think the fact that we have confirmation saying the save was successful should be enough signal to the user. Unless it's solving a user problem we've seen before - even then, I'd wonder if maybe the confirmation needs to be more obvious instead?

Reply:
Yeah, I think one of the main problems is that the confirmation is not very obvious. It's not using any color or iconography, so you just have to notice that the text changed.

Reply:
I just removed the delay in d2b5fc6cf, but going to punt on improving the confirmation.